### PR TITLE
Fix CI new build warnings

### PR DESCRIPTION
**Pull request by tcharding**
The latest nightly toolchain introduced a whole bunch of new warnings and errors, mostly to do with import statements - fix them all.

### CODE REVIEW

File: bitcoin/embedded/src/main.rs [hunk 1]
```diff
@@ -7,7 +7,6 @@ extern crate alloc;
 extern crate bitcoin;
```
Comment (Kixunil) [inline bitcoin/embedded/src/main.rs:7]: FTR these are not clippy warnings but rustc warnings. Please don't bother with fixing the commit message, this is blocking other work.

**Review by Kixunil — APPROVED** [review]
ACK 698ff6ab5f176d45190d14afc26bf4f4ef2652a7

**Comment by tcharding** [comment]
> FTR these are not clippy warnings but rustc warnings. 

I was getting them by running `cargo +nightly clippy`?

**Comment by tcharding** [comment]
Can squash once CI passes, these turned out to be way more painful than I first expected.

**Comment by tcharding** [comment]
Sorry its my bed time, can't wait for CI to run.

**Comment by Kixunil** [comment]
> I was getting them by running `cargo +nightly clippy`?

`clippy` also shows rustc warnings.

**Comment by apoelstra** [comment]
In 6e1a15f291e8a12a41b0d91f03aa779c670b4a4b there is some trailing whitespace added. Otherwise utACK 6e1a15f291e8a12a41b0d91f03aa779c670b4a4b. Please squash into fewer than 18 commits so my poor test server has less work to do :).

**Comment by tcharding** [comment]
> Otherwise utACK https://github.com/rust-bitcoin/rust-bitcoin/commit/6e1a15f291e8a12a41b0d91f03aa779c670b4a4b. Please squash into fewer than 18 commits so my poor test server has less work to do :)

5 patches, hopefully that happy place where your server fans aren't wuring too much and neither is your brain.

File: bitcoin/src/sign_message.rs [hunk 2]
```diff
@@ -165,8 +164,8 @@ mod message_signing {
         use base64::prelude::{Engine as _, BASE64_STANDARD};
 
         use super::*;
-        use crate::prelude::String;
-
+        use crate::prelude::*;
+
```
Comment (apoelstra) [inline bitcoin/src/sign_message.rs:168]: In 16f75903ea99c88e5e1f8a792cd6c0342b191936:

Trailing whitespace.

Comment (tcharding) [reply bitcoin/src/sign_message.rs:168]: still!

Comment (tcharding) [reply bitcoin/src/sign_message.rs]: This PR has non-building patches in it, we aren't enforcing green CI on every patch yet are we?

Comment (tcharding) [reply bitcoin/src/sign_message.rs]: If we are I'll have to squash it into a single patch.

**Comment by tcharding** [comment]
I squashed the whole thing down to a single patch, we don't want to have to iterate on this again while I'm asleep so it seems safer to have it as a single patch.

**Comment by tcharding** [comment]
Ouch, `just sane` returns errors on this branch but CI passes!

File: bitcoin/src/sign_message.rs (hunk 2 above)
Comment (apoelstra) [reply bitcoin/src/sign_message.rs]: We are not enforcing green CI on every patch, no, and I don't think it'll pass.

We are enforcing that *my* CI passes on every patch but I can poke at that to temporarily disable or pin nightly.

**Review by apoelstra — APPROVED** [review]
ACK 9187bf3a659860ef9d5e4bfb1b91e9413d976246

**Comment by apoelstra** [comment]
Gonna merge with the one-ACK carve-out because these are all uncontroversial non-functional changes. The majority are just rearranging private imports.

**Pull request merged**